### ISSUE TITLE
Review and cleanup variables

### ISSUE THREAD

**Issue by xak** [SAP/techne#591]
As a developer, I would like fewer and more distinct variable definitions so that I can accomodate white-labeling and theming options.

The variables defined in the current version of Techne is un-manageable and cannot be themed in the current state. As is, we would have to map, for example, `@default-greyblue` to another color, probably not blue or grey. 

Colors should be defined for their use and not their actual color so that `@color-text` could be blue or black or red with no need to change the variable name.

https://github.com/SAP/techne/blob/develop/src/less/globals/themes/_techne.less

It is recommended that variables be named related to the CSS rule for which they are most commonly used (using a BEM-like notation). For example, you may write a rule like `border-color: $border-color--table;` or `color: $color--alert`.

Basically, we need an audit of the entire variables file with recommendations for cleanup and consolidation.

**Comment by joseegm** [comment]
@xak 
It is already been done that way for the variable names. Take a look at

https://github.com/SAP/techne/blob/feature/CCE-v2.0/src/techne/system/colors.scss

and

https://github.com/SAP/techne/blob/feature/CCE-v2.0/src/techne/system/spacing.scss

Those are generic names. 

Will like to avoid the double dash "--". Don't bring any clarity.

**Theming**: What is still not in place is the "theming" functionality. If you can jump in and implement that on the POC it will be nice, you are the most experienced with SASS from all us :)

Audit... we are not porting everything from v1.5.x to v2.0. So we can skip that and create the new variables from scratch as we need it.

**Comment by joseegm** [comment]
Theming will be addressed on #594
